### PR TITLE
Product Image Gallery

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR adds the Product Image Gallery.

## How it works? 
The gallery is lazily included into the page when the page is being displayed on the desktop and is not server side rendered to avoid spending precious mobile CPU cycles on rendering a `display: none` object

## How to test it?
[marinbrasil](https://github.com/vtex-sites/marinbrasil.store/pull/171)
[storecomponents](https://github.com/vtex-sites/storecomponents.store/pull/370)

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

File: packages/gatsby-theme-vtex/src/components/ProductImageGallery/Miniatures/Page.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  item: Item
+  variant: string
+}
+
+const Page: FC<Props> = ({ item, variant }) => {
+  if (item.type === 'image') {
+    return <AspectImage ratio={1} {...item.props} {...item.props.targetProps} />
+  }
+
+  return <YoutubeThumb src={item.props.src} variant={variant} />
```
Comment (lucis) [inline packages/gatsby-theme-vtex/src/components/ProductImageGallery/Miniatures/Page.tsx:16]: Add a todo to turn this into `VideoThumb`. Our Admin supports adding videos from Vimeo, for example.

That's why I've created the VideoIframe wrapper (despite only supporting Youtube now)

File: packages/gatsby-theme-vtex/src/components/ProductImageGallery/Miniatures/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+const Miniatures: FC<Props> = ({
+  variant,
+  onSelect,
+  allItems,
+  selectedPage,
+}) => (
+  <>
+    {allItems.map((item, index) => (
+      <Box
```
Comment (lucis) [inline packages/gatsby-theme-vtex/src/components/ProductImageGallery/Miniatures/index.tsx:22]: Should them be buttons/tabable?

**Review by lucis — APPROVED** [review]

File: packages/gatsby-theme-vtex/src/components/ProductImageGallery/Miniatures/Page.tsx (hunk 1 above)
Comment (tlgimenes) [reply packages/gatsby-theme-vtex/src/components/ProductImageGallery/Miniatures/Page.tsx:16]: Cool ! I'm just not sure if we should suport it via props or via shadowing. I think shadowing is better, but I really don't know. I will add the comment though

File: packages/gatsby-theme-vtex/src/components/ProductImageGallery/Miniatures/index.tsx (hunk 2 above)
Comment (tlgimenes) [reply packages/gatsby-theme-vtex/src/components/ProductImageGallery/Miniatures/index.tsx:22]: Nice ! I think so

**Pull request merged**